product_uom_updatable/models/product_template.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Acsone SA/NV
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from functools import reduce
+from itertools import groupby
+
+from psycopg2 import sql
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    def write(self, vals):
+        uom_id = vals.pop("uom_id", False)
+        uom_po_id = vals.pop("uom_po_id", False)
+        if uom_id:
+            self._update_uom(uom_id, "uom_id")
+        if uom_po_id:
+            self._update_uom(uom_po_id, "uom_po_id")
+        res = super().write(vals)
+        return res
+
+    def _update_uom(self, uom_id, field_name):
+        uom_obj = self.env["uom.uom"]
+        sorted_items = sorted(self, key=lambda r: r[field_name])
+        for key, products_group in groupby(sorted_items, key=lambda r: r[field_name]):
+            products = reduce(
+                lambda x, y: x | y, products_group, self.env["product.template"]
+            )
```

Comment:
@glitchov It's not a good idea to use the operator '|' to merge list of recordset. It's less performant than the previous implementation ` product_ids = [product.id for product in products_group]` and far more difficult to read.

Reply:
I need the recordset to invalidate after the update and i thought it was easier to get the ids from recordset than having the ids and re-browse for recordset.